merlin/models/tf/transforms/sequence.py

```diff
@@ -62,3 +71,281 @@ def call_outputs(
+                    new_input_shapes[k] = tf.TensorShape([v[1][0], None])
+                else:
+                    # Reducing 1 position of the seq length
+                    new_input_shapes[k] = tf.TensorShape([v[0], v[1] - 1])
+
+        return new_input_shapes
+
+    def get_config(self):
+        """Returns the config of the layer as a Python dictionary."""
+        config = super().get_config()
+        config["target"] = self.target
+
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        """Creates layer from its config. Returning the instance."""
+        config = tf_utils.maybe_deserialize_keras_objects(config, ["pre", "post", "aggregation"])
+        config["schema"] = schema_utils.tensorflow_metadata_json_to_schema(config["schema"])
+        schema = config.pop("schema")
+        target = config.pop("target")
+        return cls(schema, target, **config)
+
+
+@Block.registry.register_with_multiple_names("seq_predict_next")
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class SeqPredictNext(SequenceTransform):
+    """Prepares sequential inputs and targets for next-item prediction.
+    The target is extracted from the shifted sequence of item ids and
+    the sequential input features are truncated in the last position.
+
+    Parameters
+    ----------
+    schema : Schema
+        The schema with the sequential columns to be truncated
+    target : Union[str, Tags, ColumnSchema]
+        The sequential input column that will be used to extract the target
+    pre : Optional[BlockType], optional
+        A block that is called before this method call().
+        If not set, the ListToRagged() block is applied to convert
+        the tuple representation of sequential features to RaggedTensors,
+        so that the tensors sequences can be shifted/truncated
+    """
+
+    def call(
+        self, inputs: TabularData, targets=None, training=False, testing=False, **kwargs
+    ) -> Union[Prediction, Tuple]:
+        self._check_seq_inputs_targets(inputs)
+
+        # Shifts the target column to be the next item of corresponding input column
+        new_target = inputs[self.target_name][:, 1:]
+        if targets is None:
+            targets = new_target
+        elif isinstance(targets, dict):
+            targets[self.target_name] = new_target
+        else:
+            targets = dict()
+            targets[self.target_name] = new_target
+
+        new_inputs = dict()
+        for k, v in inputs.items():
+            if k in self.schema.column_names:
+                # Removes the last item of the sequence, as it belongs to the target
+                new_inputs[k] = v[:, :-1]
+            else:
+                new_inputs[k] = v
+
+        return (new_inputs, targets)
+
+
+@Block.registry.register_with_multiple_names("seq_predict_last")
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class SeqPredictLast(SequenceTransform):
```

Comment:
For consistency in the API, we might want to pick one of  `Seq` or `Sequence` across all the layers that refer to this concept. Perhaps the longer version?

Reply:
Makes sense @oliverholworthy . Have renamed the prefix `Seq` to `Sequence`